tests/Doctrine/Migrations/Tests/Tools/Console/Command/DoctrineCommandTest.php

```diff
@@ -72,4 +73,25 @@ protected function execute(InputInterface $input, OutputInterface $output) : int
             ['interactive' => false]
         );
     }
+
+    public function testDependencyFactory() : void
+    {
+        $exceptionThrown = false;
+
+        $dependencyFactory = $this->createMock(DependencyFactory::class);
+
+        try {
+            new class ($dependencyFactory) extends DoctrineCommand
+            {
+                protected function configure() : void
+                {
+                    $this->getDependencyFactory();
+                }
+            };
+        } catch (DependenciesNotSatisfied $exception) {
+            $exceptionThrown = true;
```

Comment:
If not having an Exception is the expected result, I think you don't need the whole try-catch construct. It's true that the fail-message is expressive, but you can put the expected behaviour in the name of the test method like `testDependencyFactoryIsInitialized()` for an also expressive message on a failing test.

Reply:
So do you suggest to drop the assertion completely? My original attempt was to have no assertion and just let the exception trigger the error. But that would be a test without any assertion (PHPUnit seems to not like those).

Then I tried an assertion at the end of it with `assertTrue(true)`, but static analysis was not happy about it. It said, this assertion would always be true.

So that's what lead to the decision to use a try-catch-contraption. If you have a better way, I will gladly implement it.

Reply:
You could assert that the `db-configuration` option is available in the command when a dependency factory is provided.

> I consider this a breaking change, since this can potentially lead to one less input option for a doctrine command, thus changing the command definition and breaking scripts that provide this option.

:thinking: your change can only result in that option being available, so, one more input option, not one fewer… I don't understand

Reply:
> You could assert that the db-configuration option is available in the command when a dependency factory is provided.

This would also test the availability of the `db-configuration` option depending on the dependency factory.